server/models/crime.js

```diff
@@ -1,33 +1,68 @@
 var db = require('../../database/index');
 var Crime = require('../../database/models/Crime');
 
-var storeOpenData = (crimeData, callback) => {
-
-  crimeData.forEach((crime) => {
-    Crime.create({
-      address: crime.address,
-      category: crime.category,
-      date: crime.date,
-      dayofweek: crime.dayofweek,
-      descript: crime.descript,
-      incidntnum: crime.incidntnum,
-      location: {
-        type: "Point",
-        coordinates: [
-          crime.location.latitude, crime.location.longitude
-        ]
-      }
-    }, function (err, crime) {
-      if (err) {
-        callback(err);
-      }
+var storeOpenData = (crimeData) => {
+
+  var asyncStore = crimeData.map((crime) => {
+    return new Promise((res, rej) => {
+
+      var long = crime.location.longitude - 0;
+      var lat = crime.location.latitude - 0;
+      var mutilplier = 1000000000000;
+      var boxPadding = 20000000;
+      
+      var coords = {
+        upperLeft: [(((long * mutilplier) - boxPadding) / mutilplier), (((lat * mutilplier) + boxPadding) / mutilplier)],
+        upperRight: [(((long * mutilplier) + boxPadding) / mutilplier), (((lat * mutilplier) + boxPadding) / mutilplier)],
+        lowerRight: [(((long * mutilplier) + boxPadding) / mutilplier), (((lat * mutilplier) - boxPadding) / mutilplier)],
+        lowerLeft: [(((long * mutilplier) - boxPadding) / mutilplier), (((lat * mutilplier) - boxPadding) / mutilplier)],
+      }; 
+
+      Crime.create({
+        address: crime.address,
+        category: crime.category,
+        date: crime.date,
+        dayofweek: crime.dayofweek,
+        descript: crime.descript,
+        incidntnum: crime.incidntnum,
+        location: {
+          type: "Point",
+          coordinates: [
+            long, lat
+          ]
+        },
+        box: {
+          type: "Polygon",
+          coordinates: [[
+            coords.upperLeft,
+            coords.upperRight,
+            coords.lowerRight,
+            coords.lowerLeft,
+            coords.upperLeft
+          ]]
+        },
+        index: "box"
+      }, function (err, crime) {
+        if (err) {
+          rej(err);
+        } else {
+          res();
+        }
+      });
     });
   });
-}
+  Promise.all(asyncStore);
```

Comment:
we don't really do anything after the information is stored in openDataCaller, so it looks good! It would probably be useful to make the openDataCaller request into a function in the future so that we can call it specifically and know once it is done storing the data, but that can just go into the backlog.